src/keyboard/mod.rs

```diff
@@ -625,3 +657,311 @@ where
+                    let utf8 = if state.compose_feed(sym)
+                        != Some(ffi::xkb_compose_feed_result::XKB_COMPOSE_FEED_ACCEPTED)
+                    {
+                        None
+                    } else if let Some(status) = state.compose_status() {
+                        match status {
+                            ffi::xkb_compose_status::XKB_COMPOSE_COMPOSED => {
+                                state.compose_get_utf8()
+                            }
+                            ffi::xkb_compose_status::XKB_COMPOSE_NOTHING => state.get_utf8_raw(key),
+                            _ => None,
+                        }
+                    } else {
+                        state.get_utf8_raw(key)
+                    };
+                    let modifiers = state.mods_state.clone();
+
+                    if key_state == wl_keyboard::KeyState::Pressed {
+                        let repeatable = unsafe {
+                            if (XKBH.xkb_keymap_key_repeats)(state.xkb_keymap, key + 8) == 1 {
+                                true
+                            } else {
+                                false
+                            }
+                        };
+                        match key_repeat_kind {
+                            KeyRepeatKind::Fixed { .. } | KeyRepeatKind::System { .. } => {
+                                if repeatable {
+                                    key_held = Some(key);
+                                    // Clone variables for the thread
+                                    let thread_kill_chan = kill_chan.clone();
+                                    let thread_repeat_impl = repeat_impl.clone();
+                                    let repeat_timing = match key_repeat_kind {
+                                        KeyRepeatKind::Fixed { rate, delay, .. } => (rate, delay),
+                                        KeyRepeatKind::System { .. } => {
+                                            *system_repeat_timing.lock().unwrap()
+                                        }
+                                        _ => (0, 0),
+                                    };
+                                    user_impl.receive(
+                                        Event::Key {
+                                            serial,
+                                            time,
+                                            modifiers,
+                                            rawkey: key,
+                                            keysym: sym,
+                                            state: key_state,
+                                            utf8: utf8.clone(),
+                                        },
+                                        proxy,
+                                    );
+                                    // Start thread to send key events
+                                    thread::spawn(move || {
```

Comment:
So, here you spawn a new thread on keypress.

How does it behave if two repeating keys are pressed simultaneously?

Like, if I press `a`, it starts repeating. Then I press `z` without releasing `a`, what happens?

What my current DE does is that the previously repeating key stops repeating, and the new one is now the only one repeating.

Reply:
My previous logic for this was wrong (I did have it at one time but must of removed it). Now it checks that no other key is being held before it sends any new key events. If you want I can change this so it sends key events but doesn't repeat them if another key is being pressed instead.

Reply:
Or I can make it can stop all key repetition (kill all threads) or kill the previous key repetition and just repeat the new key, its up to you.

Reply:
Well, the behavior I've seen everywhere is that only the last pressed key repeats, so I'd say the best is probably to do the same thing.

Reply:
Nah this should hopefully only require an small adjustment to make state be passed by arc instead of pure clone.